starter-core/src/main/java/io/micronaut/starter/feature/dekorate/DekorateFeature.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.dekorate;
+
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.OneOfFeature;
+
+/**
+ * @author Pavol Gressa
+ * @since 2.0
+ */
+public interface DekorateFeature  extends OneOfFeature {
```

Comment:
This interface is here in order to support OneOfFeature. The reason is that from usability POV you want to have one of:
- Kubernetes 
- Knative
- Openshift
- Halkyon

All of them are on top of K8s but are generating different K8s manifests. Question is whether it makes sense to force users to choose one of these.

Reply:
(I don't know about these tools)

Are these four exclusive or a user could add manually the necessary dependencies for the four of them and they all can work at the same time? Or maybe they can work at the same time but it doesn't make sense. 
 In case they can work together, then they shouldn't be a `OneOfFeature`.